phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3604,6 +3681,18 @@ public MutationState addColumn(PTable table, List<ColumnDef> origColumnDefs,
             }
         } finally {
             connection.setAutoCommit(wasAutoCommit);
+            if (acquiredMutex && !columns.isEmpty()) {
```

Comment:
where is acquiredMutex set to true?

Reply:
@ankitsinghal  Sorry, I didn't update the PR with all my latest changes, I fixed this. Can you please review?